spec/importers/file_importers/educator_section_assignments_importer.rb

```diff
@@ -133,5 +133,21 @@
           expect(EducatorSectionAssignment.count).to eq(1)
         end
       end
+
+      context 'delete only stale assignments from schools being imported' do
+        before do
+          FactoryGirl.create_list(:educator_section_assignment,20)
+          FactoryGirl.create(:educator_section_assignment, educator_id: educator.id, section_id: section.id)
+
+          esa_importer = described_class.new
+          esa_importer.school_scope = ['SHS']
+          esa_importer.import_row(row)
+          esa_importer.delete_rows
+        end
+
+        it 'deletes all student section assignments for this school except the recently imported one' do
+          expect(EducatorSectionAssignment.count).to eq(21)
```

Comment:
I'm confused - this looks to me like it creates 20 records, then a 21st, and then it imports a row and deletes in the test.  I can't tell what to expect to happen - is the idea that the record on line 140 is stale, or can you help me understand?

Reply:
Yeah, it's a little confusing the way I added it in for school specific. The original test, doesn't test school scope at all, so it defaults to all schools. Create 20 random assignments, create 1 specific one. The one specific one is imported. All the others are stale since they weren't imported.

The new test, strikingly similar but with a school scope added won't touch the 20 random assignments because they are for a random school and the importer's `school_scope` is set to SHS.

Reply:
Are the 20 assignments literally random with what school they refer to?  That seems like it would lead to non-deterministic test runs if one happened to be in SHS.

To clarify, if it's simple the generator could make 20 assignments with the school to be FC so it's clear in the test that those records shouldn't be deleted, or the expectation could check that "there's 20 non-SHS records" and "there's only 1 SHS records since it was updated."

Reply:
Agreed, will make more explicit. Actually, the school created by the factory would just have null for the local_id. I'll make it explicitly create them in another school.